velox/dwio/parquet/writer/Writer.cpp

```diff
@@ -43,7 +43,11 @@ void Writer::write(const RowVectorPtr& data) {
             arrowProperties));
   }
 
-  PARQUET_THROW_NOT_OK(arrowWriter_->WriteRecordBatch(*recordBatch));
+  if (changeWriteAPI) {
+    PARQUET_THROW_NOT_OK(arrowWriter_->WriteRecordBatch(*recordBatch));
+  } else {
+    PARQUET_THROW_NOT_OK(arrowWriter_->WriteTable(*table));
```

Comment:
Why not using the same API? [WriteRecordBatch](https://github.com/apache/arrow/blob/main/cpp/src/arrow/adapters/orc/adapter.cc#L817-L820) underlyingly will create a Arrow table to call ```WriteTable```. The only difference is about whether passing in the size parameter.

Reply:
Here is [FileWriter#WriteTable()](https://github.com/apache/arrow/blob/7432cc7fa4d275d6e0f57b672a0677358fc538a6/cpp/src/parquet/arrow/writer.cc#L359) and [FileWriter#WriteRecordBatch()](https://github.com/apache/arrow/blob/7432cc7fa4d275d6e0f57b672a0677358fc538a6/cpp/src/parquet/arrow/writer.cc#L403). We test the above two write API in 1gb TPC-DS store_sales table. The FileWriter#WriteTable() need cost 800s and the WriteRecordBatch only need 4.9s. It seems the WriteRecordBatch use buffered RowGroup Writer to improve performance.

Reply:
Per discussed, a few suggestions: 1. TODO; 2. Avoid magic number; 3. optimize parameter initialization by moving table parameter within L48 else block.

Reply:
Thanks for your review. No need to modify the Write API anymore. The root cause is that reallocate will be called every time when DataBuffer reserve, causing serious memcpy. Here we adjust the size of reallocate to 2x.

Reply:
Great!